extensions/amp-story/1.0/utils.js

```diff
@@ -323,3 +323,22 @@ export function triggerClickFromLightDom(anchorElement, domElement) {
   outerAnchor.click();
   outerAnchor.remove();
 }
+
+/**
+ * Makes a proxy URL if document is served from a proxy origin. No-op otherwise.
+ * @param {string} url
+ * @param {!Element} pageEl
+ * @return {string}
+ */
+export const maybeMakeProxyUrl = (url, pageEl) => {
+  const urlService = Services.urlForDoc(pageEl);
+  const loc = pageEl.getAmpDoc().win.location;
+  if (!urlService.isProxyOrigin(loc.origin) || urlService.isProxyOrigin(url)) {
+    return url;
+  }
+  const resolvedRelativeUrl = urlService.resolveRelativeUrl(
+    url,
+    urlService.getSourceOrigin(loc.href)
+  );
+  return loc.origin + '/i/s/' + resolvedRelativeUrl.replace(/https?:\/\//, '');
```

Comment:
These URLs support a bunch of useful parameters that we should add to the URL, like max width etc. Please check the "URL path" documentation here: https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cache-urls/

I think specifying a width here would be useful and save some bytes

Reply:
Dynamic resize is expensive, and sacrifices AVIF encoding (because that's too expensive to do on the fly). Using a regular image URL will likely be faster due to better caching, faster response times, and better encoding formats. Using an oversized image is a publisher fault.

Reply:
Definitely using oversized images are not great for components that show the images in small regions. When passing these parameters do the images get resized every time, or would they only be resized on the first request to that resource at that resolution? I think for these types of images we'd only want to ask for the small image if we can, and not the original sized asset

Reply:
> When passing these parameters do the images get resized every time, or would they only be resized on the first request to that resource at that resolution? 

They're resized every time. The persistence layer never stores resizes, only originals. The local-cache _may_ keep an resized item, but it's an extremely small LRU competing with literal billions of other responses.

> I think for these types of images we'd only want to ask for the small image if we can, and not the original sized asset

What size is the original asset? Is the publisher aware this image is being shown in a small area?

Reply:
Thank you for chiming in Justin, let's keep `/i/s` and no resizing then.